File: src/main/scala/com/github/kazuhito_m/twitterbattler/primitive/presentation/controller/BattlerController.scala
package com.github.kazuhito_m.twitterbattler.primitive.presentation.controller

import java.security.Principal

import com.github.kazuhito_m.twitterbattler.primitive.application.BattlerService
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.web.bind.annotation.RequestMethod._
import org.springframework.web.bind.annotation.{RequestMapping, RestController}

@RestController
@RequestMapping(Array("/api/battler"))
class BattlerController(battlerService: BattlerService) {

  protected val log: Logger = LoggerFactory.getLogger(classOf[BattlerController])

  @RequestMapping(value = Array("getPlayer"), method = Array(GET))
  def getPlayer(user: Principal) = battlerService.getPlayer(user.getName)

}

File: server/src/main/java/com/github/kazuhito_m/twitterbattler/primitive/presentation/controller/battler/BattlerController.java
package com.github.kazuhito_m.twitterbattler.primitive.presentation.controller.battler;

import com.github.kazuhito_m.twitterbattler.primitive.application.service.battler.BattlerService;
import com.github.kazuhito_m.twitterbattler.primitive.domain.model.battler.Battler;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.security.Principal;

@RestController
@CrossOrigin
@RequestMapping("/api/battler")
public class BattlerController {
    final BattlerService battlerService;

    @GetMapping("getPlayer")
    Battler getPlayer(Principal user) {
        return battlerService.getPlayer(user.getName());
    }

    BattlerController(BattlerService battlerService) {
        this.battlerService = battlerService;
    }
}
